akka-projection-core/src/main/scala/akka/projection/scaladsl/SourceProvider.scala

```diff
@@ -26,6 +27,46 @@ trait SourceProvider[Offset, Envelope] {
 
 }
 
+/**
+ * By default, a `SourceProvider` uses the stored offset when starting the Projection. This offset can be adjusted
+ * by defining the `startOffset` function.
+ */
+trait CustomStartOffsetSourceProvider[Offset, Envelope] extends SourceProvider[Offset, Envelope] {
```

Comment:
Would it be possible to add it directly to `SourceProvider` with default implementation instead of introducing another trait? Compatibility concern could be that we change it from a pure interface to trait with impl, and that the method names may clash with existing methods in user's implementation of SourceProvider (although unlikely).